app/utils/lava_log_parser.py

```diff
@@ -122,27 +124,33 @@ def run(log, boot, txt, html):
 
     start_ts = None
     log_buffer = []
+    lineno = 0
 
     for line in log:
+        lineno = lineno + 1
         dt, level, msg = (line.get(k) for k in ["dt", "lvl", "msg"])
         raw_ts = DT_RE.match(dt).groups()[1]
         timestamp = "<span class=\"timestamp\">{}  </span>".format(raw_ts)
+        sl = "<p id=%d>" % lineno
```

Comment:
Maybe use another type of HTML element instead of `<p>` with zero margin.  Paragraphs are supposed to have space between them.  The log could be a list and each line an item, or you could use `<span>` which doesn't add any formatting.

Reply:
Also, using just the plain number as the `id` value may not be very wise as these are global identifiers for the document.  Rather than `id=123`, maybe do like LAVA does with `id="L123"` or something even more explicit such as `id="line-123"`.